packages/core/src/shorthand-parser/index.ts

```diff
@@ -292,3 +293,46 @@ export const boxShadow = (tokens: any, value: string) => {
       .join(', '),
   };
 };
+
+export const textDecoration = createPropertyParser((tokens: any, css: any, value: any) => {
+  if (matchString(value, /unset/)) {
+    // we are checking the existing value here because we want to retain the long hand property that was previously applied
+    /**
+     * For example:
+     * textDecorationColor: red
+     * textDecoration: unset
+     *
+     * The above should result in a final css rule like the following
+     * text-decoration-style: unset;
+     * text-decoration-line: unset;
+     * text-decoration-color: red
+     *
+     * This is as per the css rules
+     */
+    css.textDecorationStyle = css.textDecorationStyle || value;
+    css.textDecorationLine = css.textDecorationLine || value;
+    css.textDecorationColor = css.textDecorationColor || value;
+  } else if (matchString(value, /solid|double|dotted|dashed|wavy/)) {
+    css.textDecorationStyle = value;
+  } else if (value.match(/none|underline|overline|line-through|blink/)) {
+    css.textDecorationLine = setChainedValue(css.textDecorationLine, value, ' ');
+  } else {
+    css.textDecorationColor = tokens.colors[value] || value;
```

Comment:
We need to handle `text-decoration-thickness` in here as well and then add a test case that includes parsing it

Reply:
text-decoration-thickness seem to be only supported on firefox and safari at the moment. Do we still need to add it?
https://caniuse.com/?search=text-decoration-thickness